### PR TITLE
Fix for Koajs v1 & v2

### PR DESCRIPTION
**Pull request by basickarl**
Tried on both versions.

Added a simple if statement.
Added information to the README.md.

Related issue:
https://github.com/pillarjs/send/issues/118

### CODE REVIEW

**Comment by basickarl** [comment]
@dougwilson Awesome! I agree, it's an ugly fix haha. Sure no problem, get back when you have time so we can work on this.^^ 

Indeed POC! Wanted you to see how it would work in koa.
Ah did I mess up the format? (I did run the eslint, didn't get any errors, or are you meaning something else?).
Yea need to write tests as well, I do agree.

File: index.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ var listenerCount = EventEmitter.listenerCount ||
  * @public
  */
 
-function send (req, path, options) {
+function send(req, path, options) {
```
Comment (jcready) [inline index.js]: Please remove all of these changes where you're only adding or removing white space as it only clutters the diff and adds nothing of value.

File: index.js [hunk 2]
```diff
@@ -734,25 +734,22 @@ SendStream.prototype.sendIndex = function sendIndex (path) {
...
-SendStream.prototype.stream = function stream (path, options) {
+SendStream.prototype.stream = function stream(path, options) {
   // TODO: this is all lame, refactor meeee
   var finished = false
   var self = this
   var res = this.res
 
   // pipe
   var stream = fs.createReadStream(path, options)
-  this.emit('stream', stream)
```
Comment (jcready) [inline index.js]: Why not emit the stream here?

File: index.js [hunk 3]
```diff
@@ -764,9 +761,18 @@ SendStream.prototype.stream = function stream (path, options) {
...
-    self.emit('end')
+  this.emit('stream', stream)
+
+  // pipe and end stream manually if set
+  if (options.skipStreamPipe)
+    return;
+
+  stream.pipe(res)
+
+  // response finished, done with the fd
```
Comment (jcready) [inline index.js]: Why isn't the `onFinished` handler (which will clean up the stream automatically for you) being added when `skipStreamPipe` is true?

File: package.json [hunk 4]
```diff
@@ -53,4 +54,4 @@
     "test-ci": "istanbul cover node_modules/mocha/bin/_mocha --report lcovonly -- --check-leaks --reporter spec",
     "test-cov": "istanbul cover node_modules/mocha/bin/_mocha -- --check-leaks --reporter dot"
   }
-}
+}
```
Comment (jcready) [inline package.json]: Restore the empty line at the end of the file.

**Comment by dougwilson** [comment]
It looks like nothing has ben followed up on any of he code reviews, so I'm going to close this. Please feel free to make a new PR when you have time to update your code :) !

**Pull request closed without merging**